content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,292 @@
+
+### Pod priority and preemption
+
+Pod preemption is a Kubernetes feature that allows the cluster to [preempt](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption) lower-priority pods over high-priority pods in the event that there aren't enough resources to run all current Pods. To learn more about this concept, please refer to [Pod Priority and Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption/) in the Kubernetes documentation.
+
+Pod priority and preemption are Kubernetes features:
+- Which help in deciding a priority while scheduling pods.
+- Which check if the mission-critical pods are up and running and are not evicted.
+- Decide how the cluster will behave in the event of a resource crunch.
+- Decide which lower priority pods are preempted (evicted) from the nodes when there are resource crunch and high priority pods are available in the queue for scheduling.
+
+### PriorityClass requirements
+
+Before you set up PriorityClasses, there are a few things to consider.
+
+1. Decide on which  PriorityClasses are needed.
+   Example: Based on environments, or based on the kind of pods, or based on the kind of applications.
+2. Which PriorityClass resource will be the default for the cluster (otherwise the pods without a `priorityClassName` will be treated as priority 0).
+3. Respective PriorityClass resources in the cluster based on the decision made on #1.
+4. How you'll name your priority classes.
+5. How you'll make sure that Pods for your workloads are running with the right PriorityClass.
+
+## PriorityClass hands-on example
+
+Let’s say there are 3 application pods: one for prod, one for preprod, and one for development. Below are three sample YAML manifest files for each of those.
+
+```yaml
+---
+# development
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dev-nginx
+  labels:
+    env: dev
+spec:
+  containers:
+  - name: dev-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "256Mi"
+         cpu: "0.2"
+      limits:
+        memory: ".5Gi"
+        cpu: "0.5"
+```
+
+```yaml
+---
+# preproduction
+apiVersion: v1
+kind: Pod
+metadata:
+  name: preprod-nginx
+  labels:
+    env: preprod
+spec:
+  containers:
+  - name: preprod-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "1.5Gi"
+        cpu: "1.5"
+      limits:
+        memory: "2Gi"
+        cpu: "2"
+```
+
+```yaml
+---
+# production
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prod-nginx
+  labels:
+    env: prod
+spec:
+  containers:
+  - name: prod-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "2Gi"
+        cpu: "2"
+      limits:
+        memory: "2Gi"
+        cpu: "2"
+```
+
+Now, I'll create each of these pods by `kubectl create -f <FILE.yaml>` command, then check their status by using `kubectl get pods`. I can see if they are up and ready to serve traffic:
```

Comment:
Kindly refrain from the usage of first-person pronouns.

Reply:
Sure, I will review this article again and will update it accordingly.

Reply:
Hi @divya-mohan0209 , 

I appreciate your time here, I have made changes based on all your recommendations.

Requesting you to please re-review the changes.

The preview version can be found [here.](https://deploy-preview-37713--kubernetes-io-main-staging.netlify.app/blog/2022/12/10/protect-mission-critical-pods-priorityclass/)

Please let me know if further changes are needed.

CC: @sftim 

Thank you,
Sunny